packages/core/src/types/types.ts

```diff
@@ -85,3 +85,54 @@ export type ServiceError = {
 	name: string;
 	message: string;
 };
+
+export type ResourceConfig = {
+	API?: {};
+	Analytics?: {};
+	Auth?: {
+		userPoolId?: string;
+		identityPoolId?: string;
+		userPoolWebClientId?: string;
+	};
+	DataStore?: {};
+	Interactions?: {};
+	Notifications?: {};
+	Predictions?: {};
+	Storage?: {};
+};
+
+
+export type LibraryOptions = {
+	Auth?: {
+		tokenProvider?: TokenProvider;
+		refreshTokenClient?: TokenRefreshClient;
+		credentialsProvider?: CredentialsProvider;
+		storage?: AuthStorage;
+	} | null;
```

Comment:
```suggestion
	};
```

I don't think the `null` is needed?